### DISCUSSION TITLE
Style training with new method

### DISCUSSION THREAD

**Discussion by specblades** [TheLastBen/fast-stable-diffusion#403] [Q&A]
Im writing it during the training process, but is goes pretty badly in comparison with old methods.

So.

1.How do I let the program know that I'm teaching it a style?
2.My images contantain buildings, nature and some people. Do i need to use Contain_Faces: Both or No? "No" is disabling prior preservation?
3.How much iters i need for 14 images and how much Train_text_encoder_for percents?

Please, give me short tutorial of some tips on style training with new method.

UPD.

I think i finished training.
I used: Contain_Faces: Both, 14 images, 10%, 7000 steps. 
Token is weak. To use it, you have to put it at the very beginning and set ~1.3 weight. With each new word in promt this value has to be changed.
Results are gut, i think, but i still want some tips! :)

![00067-1598478077-in (drrstl_1 3) style, beautiful busty demi rose standing in sunny meadow, wearing negligee and high heels, tarot card design, a](https://user-images.githubusercontent.com/113993746/200149851-0b0ce62f-b943-44cd-a76d-01e04b897047.png)
![00075-2601538226-in (drrstl_1 3) style, A frowning detective looks at the crime scene, a dead body, the area cordoned off with tape, tarot card d](https://user-images.githubusercontent.com/113993746/200149853-461618fc-e500-4af1-b222-6542209e1a98.png)
![00081-4248023648-in (drrstl_1 3) style, A frowning detective looking on a dead body, the crime scene, the area cordoned off with tape, deep night](https://user-images.githubusercontent.com/113993746/200149856-d2b35920-a4a7-40e6-87e4-7d5833af0a85.png)
![00096-2563217602-in (drrstl_1 3) style, an old detective in hat in front of a (burning church), a crescent moon in the night sky,countryside, nig](https://user-images.githubusercontent.com/113993746/200149858-1f19a65d-b866-41c7-9189-f65e618564eb.png)
![00102-3221338181-in (drrstl_1 3) style, a frog sitting in a picnic basket, in scenic environment, autumn, twilight, deathpunk, trending on behanc](https://user-images.githubusercontent.com/113993746/200149859-f19e2435-6916-4044-89c7-e4805def6a93.png)

**Comment by TheLastBen** [comment]
The text_encoder % is basically the weight of the trained subject, for a style or a subject that is trained for styling, keep the % at 10-20% at it should be fine.

How do you find the results with the new method ? (requires more complicated prompting)

**Comment by specblades** [comment]
@TheLastBen 
This time I did even better than with the old method (with reg images), but I think the reason is that I picked up the dataset better.
The current method seems to be longer and less clear - you need to experimentally find the golden value of text_encoder %. Similar to working with TI, i think. It was easy to retrain the model before, but now it seems to be the opposite.
Dotay i will try to train with 15% or 20% on another style (with realistic shadows) and be able to say more

**Reply by nawnie** [reply]
with enough trials maybe we can get the text encoder % set for us based on what we chose (style person[photo-real],                    character [stylized person]), or a filter/setting via a dropbox. 

I've done similar with the learning rates as well (used a slider here for aesthetics)

**Comment by SAC020** [comment]
Hi,

Sorry if asking noob questions, but I feel completely lost in random outcomes.

I have the following picture set: 30 full body pics + 30 face closeups of one woman, naked, in a specific bdsm stance (which is not "covered" by standard SD model vocabulary), on a white background.

I have named them "personaltoken_woman_specificbdsmstance_white_background (1)..(60)" and the session is named likewise. I have used 100 .. 150 steps x 60 pics, with a 30 .. 20% text encoding - did 2 models.

Which training parameters, prompt and cfg should I use to get:

- that specific woman performing that bdsm stance - this is pretty reliable, however I cannot get reliable background control, it usually places her in a forest while I am asking for a dungeon

- that specific woman, wearing clothes (not naked as the model) - this seems very hard to get

- that specific woman doing something else (going shopping) - this seems easy to get if I don't use the bdsm stance in the prompt, however she's naked

- another woman doing the bdsm stance - almost impossible to get

TY

**Reply by nawnie** [reply]
Um...

You seem to want to import many ideas, start small and build its vocabulary, show it what each thing is no other tags, once you can produce each item on its own then get specific. 


Or right the generate option in web ui click forever generate.. it'll get there eventually

**Reply by SAC020** [reply]
Not sure what you mean by "build its vocabulary"? How do I do that?

**Reply by nawnie** [reply]
That's what dreambooth attempts to do, learn concepts.

You could probably also find a model that will help get ya half way there.

**Reply by nawnie** [reply]
so first off, if it doesnt know what bondage is maybe show it some chains and leather straps on a person label it bondage, after it has been trained on that concept (in my own head theory) you would now be able to tag bondage on to a picture (say a certain position), and it would now be able to recognize the word bondage in its proper context. 

also note that your position, Token_person, and activity would all best be trained one after another being as you will want text encoder higher on the person then on the ... "acts"

Here check this one out, in the zip with the ckpt they include their tags and training photos, maybe this will help you get the results you are looking for.

THIS IS A LINK TO A SITE THAT HAS LINKS TO SD MODELS... most are NSFW
https://rentry.org/sdmodels#sxd-v08

**Reply by nawnie** [reply]
Lastly img2img / inpainting is your friend. doesnt matter who sd generates when you can just black their face out and ask for another.
Calling 2 people tends to break SD and cause blending to happen since they are both people and you asked for a person with luck of the seed and a few tries its possible. but with inpaint its gonna happen much quicker. 

If you havnt done so recently its time to update your webUI as there are a ton of tools that will help. Tokenizer, embedding editor, a not really working dreambooth (this(repo) is much better atm, more buttons to push and knobs to turn)

**Reply by SAC020** [reply]
Thank you, I'll continue to look into it
